wcfsetup/install/files/acp/templates/index.tpl

```diff
@@ -15,6 +15,12 @@
 	{/foreach}
 {/if}
 
+{if !$taintedApplications|empty}
+	{foreach from=$taintedApplications item=$taintedApplication}
+		<div class="error">{lang}wcf.acp.package.application.isTainted{/lang}</div>
+	{/foreach}
+{/if}
```

Comment:
```suggestion
{foreach from=$taintedApplications item=$taintedApplication}
	<div class="error">{lang}wcf.acp.package.application.isTainted{/lang}</div>
{/foreach}
```

Less noise in the template, PHP will happily iterate over an empty array.

Reply:
I've stolen this from the surrounding code (above). Will adjust all of the locations.